p2p/net/swarm/swarm_stream.go

```diff
@@ -75,9 +75,19 @@ func (s *Stream) Write(p []byte) (int, error) {
 	return n, err
 }
 
+type asyncCloser interface {
+	AsyncClose(onDone func()) error
```

Comment:
Add docs on the method please, I had to read the implementation to understand this actually also started the close process.

Reply:
Thx for explaining why it's private.
Just so you know this doesn't actually prevent anyone from implementing it.
You can use a badge using an internal package if you want to do.

Reply:
This seems totally unnecessary.  Why not just call `closeAndRemoveStream` immediately? From the perspective of libp2p, the stream is closed. If calling `closeAndRemoveStream` causes issues, that's a separate bug that needs to be fixed.


Really, _all_ stream closure is async, this isn't unique to webrtc.